quantum/action_util.h

```diff
@@ -25,6 +25,20 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 extern "C" {
 #endif
 
+typedef union {
+    uint8_t raw;
+    struct {
+        bool left_ctrl : 1;
+        bool left_shift : 1;
+        bool left_alt : 1;
+        bool left_gui : 1;
+        bool right_ctrl : 1;
+        bool right_shift : 1;
+        bool right_alt : 1;
+        bool right_gui : 1;
+    };
+} mod_t;
```

Comment:
```suggestion
} PACKED mod_t;
```

Please add a static assertion that `mod_t` is always `sizeof(uint8_t)`. It is unlikely to go oversize but better safe then sorry, especially if it will be used in the keyboard report structs.

Reply:
No problem :-). The `PACKED` attribute should be added to the union to force single byte alignment.

> Will also apply the answer to led_t, which doesn't have the attribute either

Good catch. There is also the opportunity to use the `led_t` struct down to the usb drivers which currently operate on the raw `uint8_t`: https://github.com/qmk/qmk_firmware/blob/e5466df800044b267d881f94a1791759e654f095/tmk_core/protocol/host_driver.h#L27

Reply:
Will do in a follow-up PR, same reasoning goes for updating keyboards' code to use new getter functions (where they help readability)

Going to keep this PR small, to get the feature merged as soon as possible. Cleanup/enhancement can come later :)